great_expectations/render/renderer/content_block/content_block.py

```diff
@@ -94,8 +94,15 @@ def _render_list(  # noqa: PLR0913, PLR0912
         )
         for obj_ in render_object:
             expectation_type = cls._get_expectation_type(obj_)
+            expectation_config = (
+                obj_.expectation_config
+                if isinstance(obj_, ExpectationValidationResult)
+                else obj_
+            )
 
-            content_block_fn = cls._get_content_block_fn(expectation_type)
+            content_block_fn = cls._get_content_block_fn(
+                expectation_type=expectation_type, expectation_config=expectation_config
+            )
```

Comment:
This logic is a bit convoluted to begin with - we either have an `ExpectationConfiguration` or an `ExpectationValidationResult`.

In order to find out if we have a `description`, we need to parse out the config, convert it to an `Expectation`, access the `description`, and then use it in a custom `content_block_fn`

Reply:
Multiple types in this list does make this convoluted.

Reply:
the list will be consistent in the types it contains